transport-classes-epoll/src/main/java/io/netty5/channel/epoll/NativeDatagramPacketArray.java

```diff
@@ -109,24 +107,38 @@ boolean addReadable(Buffer buf, int segmentLen, InetSocketAddress recipient) {
                 return false;
             }
             for (var c = first; c != null; c = c.nextReadable()) {
-                int writableBytes = c.readableBytes();
-                int byteCount = segmentLen == 0? writableBytes : Math.min(writableBytes, segmentLen);
-                if (iovArray.addReadable(c, byteCount)) {
-                    NativeDatagramPacket p = packets[count];
-                    long packetAddr = iovArray.memoryAddress(iovArrayStart);
-                    p.init(packetAddr, iovArray.count() - iovArrayStart, segmentLen, recipient);
-                    count++;
-                    c.skipReadableBytes(byteCount);
+                if (segmentLen == 0) {
+                    int readableBytes = c.readableBytes();
+                    if (iovArray.addReadable(c, readableBytes)) {
```

Comment:
The problem here was that we basically created one `NativeDatagramPacket` per component while we need to just only create one with hold multiple iovecs.